build/container/Dockerfile

```diff
@@ -182,7 +113,83 @@ RUN \
     \
     CGO_ENABLED=1 GOOS=linux   GOARCH=amd64 CC=x86_64-linux-gnu-gcc   go install -installsuffix race -a std
 
-WORKDIR /build
+#
+# build packages we need for ceph from source on all supported platforms
+#
+
+COPY external /build/external/
+
+RUN set -e;\
+    for a in linux_amd64 linux_arm64; do \
+      cmake -Hexternal -Bexternal/build/${a} -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_TOOLCHAIN_FILE=`pwd`/external/toolchain/gcc.${a}.cmake ;\
+      cmake --build external/build/${a} --target install;\
+    done ;\
+    rm -fr external
+
+RUN set -e;\
+    for a in amd64 arm64; do \
+    DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-overwrite" install -yq --no-install-recommends \
+    libatomic-ops-dev:${a} \
+    libexpat1-dev:${a} \
+    libkeyutils-dev:${a} \
+    libsnappy-dev:${a} \
+    libssl-dev:${a} \
+    libudev-dev:${a} \
+    ; done
+
+# Some of the packages we need for ceph are either not available in yakkety or dont support multi-arch.
```

Comment:
I'm working my way through all these packages. this whole section should go away